src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2081,7 +2081,10 @@ impl Coordinator {
         let in_immediate_multi_stmt_txn = session.transaction().is_in_multi_statement_transaction()
             && when == QueryWhen::Immediately;
 
-        check_no_invalid_log_reads(catalog, cluster, &source_ids, &mut target_replica)?;
+        let notices = check_log_reads(catalog, cluster, &source_ids, &mut target_replica)?;
+        if session.vars().enable_per_replica_notice() {
+            session.add_notices(notices);
```

Comment:
This could be misleading because it assumes all notices are related to `enable_per_replica_notice`. This is true right now, but there's nothing that prevents us from adding different notices to `check_log_reads`. Could we check the flag inside the function instead and unconditionally return notices here?